app/utils/kcidb.py

```diff
@@ -122,6 +122,19 @@ def push_build(build_id, first, bq_options, db_options={}, db=None):
         })
         bq_data['revisions'] = [bq_revision]
 
+    files = {
+        'kernel_image': build[models.KERNEL_IMAGE_KEY],
+        'modules.tar.xz': build[models.MODULES_KEY],
+        'System.map': build[models.SYSTEM_MAP_KEY],
+    }
+    output_files = []
+    for name, file in files.iteritems():
+        url = "/".join([
+            STORAGE_URL,
+            build[models.FILE_SERVER_RESOURCE_KEY],
+            file])
```

Comment:
Exception seen on staging:
```
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]: [   ERROR/MainThread] Task kcidb-build[ade04b95-7fcc-468a-9cfb-e07ba5cacab7] raised unexpected: TypeError('sequence item 2: expected string or Unicode, NoneType found',)
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]: Traceback (most recent call last):
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:   File "/srv/.venv/api.staging.kernelci.org/local/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:     R = retval = fun(*args, **kwargs)
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:   File "/srv/.venv/api.staging.kernelci.org/local/lib/python2.7/site-packages/celery/app/trace.py", line 438, in __protected_call__
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:     return self.run(*args, **kwargs)
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:   File "/srv/api.staging.kernelci.org/app/taskqueue/tasks/kcidb.py", line 31, in push_build
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:     taskc.app.conf.db_options)
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:   File "/srv/api.staging.kernelci.org/app/utils/kcidb.py", line 135, in push_build
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:     file])
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]: TypeError: sequence item 2: expected string or Unicode, NoneType found
```
So probably because one file isn't in a build.  I suppose that can easily happen with builds that fail, but I haven't debugged it to see what the data actually was.

Reply:
You were right.  On failed builds, the `file` variable will be `None` causing this to fail.  Added a check to catch that.